test/s3-file-system.js

```diff
@@ -27,7 +27,7 @@ test.before(async t => {
         }
     });
     AWS.mock( "S3", "putObject", (params, callback) => {
-        uploadedObject = params;
```

Comment:
all looks good, I just wonder why this change was needed. Could you shed some light on it?

Reply:
Ah, I thought the reason for failing tests are due to asynchronous behavior for each `test` function, so every `putObject` params store to `uploadedObjects`.

Occasionally some assertions are processed to another test object, I thought (e.g testing file is `regular.jpg` but test seems to assert to `network-error.jpg`, maybe)

Reply:
sorry, I've missed that context. That totaly make sense.